packages/glimmer-runtime/lib/compiled/opcodes/dom.ts

```diff
@@ -442,30 +441,29 @@ export class ModifierOpcode extends Opcode {
 
 export class UpdateModifierOpcode extends UpdatingOpcode {
   public type = "update-modifier";
-
   private element: Element;
   private dynamicScope: DynamicScope;
   private args: EvaluatedArgs;
   private manager: ModifierManager<Opaque>;
   private modifier: Opaque;
   private lastUpdated: Revision;
 
-  constructor({ manager, modifier, element, dynamicScope, args }: { manager: ModifierManager<Opaque>, modifier: Opaque, element: Element, dynamicScope: DynamicScope, args: EvaluatedArgs }) {
+  constructor({ manager, modifier, element }: { manager: ModifierManager<Opaque>, modifier: Opaque, element: Element }) {
     super();
     this.modifier = modifier;
     this.manager = manager;
     this.element = element;
-    this.dynamicScope = dynamicScope;
-    this.args = args;
-    this.tag = args.tag;
-    this.lastUpdated = args.tag.value();
+    this.dynamicScope = manager.dynamicScope;
+    this.args = manager.args;
+    this.tag = this.args.tag;
+    this.lastUpdated = this.args.tag.value();
   }
 
   evaluate(vm: UpdatingVM) {
-    let { manager, modifier, element, dynamicScope, args, lastUpdated } = this;
+    let { manager, modifier, element, args, lastUpdated } = this;
 
     if (!args.tag.validate(lastUpdated)) {
-      manager.update(modifier, element, args, vm.dom, dynamicScope);
+      manager.update(modifier, element, vm.env.getDOM());
```

Comment:
This should probably be async for consistency – the mental model is all modifiers run _after_ component hooks.

So this line should just be `env.didUpdateModifier` (maybe the name should be a `installModifier` and `updateModifier` – because all modifiers are async, and we didn't actually apply the update here)